02_knight_and_princess/tevfik_aksoy/main.cpp

```diff
@@ -106,15 +199,63 @@ class Graph {
     return false;
   }
 
+  VertexDepth get_max_depth() { return max_depth_; }
+  void set_max_depth(VertexDepth max_depth) { max_depth_ = max_depth; }
+  int get_new_vertices_num() { return new_vertices_num_; }
+
+  // Returns depth of vertex in the graph
+  const VertexDepth depth_of(const VertexId& id) const {
+    return vertices_[id].depth;
+  }
+
+  // Returns the number of vertices created by vertex in the graph
+  const int created_by_vertex(const VertexId& id) const {
+    if (id == 0)
+      return new_vertices_num_;
+    int count = 0;
+    for (const auto& vertex : vertices_) {
+      if (vertex.source_id == id)
+        count++;
+    }
+    return count;
+  }
+
+  vector<Vertex> get_vertices() const { return vertices_; }
+  vector<Edge> get_edges() const { return edges_; }

```

Comment:
Чем возвращать копию потенциально очень тяжелых данных, лучше вернуть `const` ссылку, а если пользователю понадобится копия - он сделает её сам
```suggestion
  const vector<Vertex>& get_vertices() const { return vertices_; }
  const vector<Edge>& get_edges() const { return edges_; }
```